### PR TITLE
feat(http): set the statusText property from the XMLHttpRequest instance

### PR DESCRIPTION
**Pull request by lifus**
It relates to #2800

### CODE REVIEW

**Comment by vsavkin** [comment]
@lifus could you rebase the PR?

File: modules/angular2/test/http/backends/xhr_backend_spec.ts [hunk 1]
```diff
@@ -333,6 +337,35 @@ export function main() {
...
+           existingXHRs[0].setStatusCode(200);
+           existingXHRs[0].dispatchEvent('load');
+         }));
+
+      it('should set the empty status text if it is not present in XMLHttpRequest instance',
+         inject([AsyncTestCompleter], (async: AsyncTestCompleter) => {
+           var connection = new XHRConnection(sampleRequest, new MockBrowserXHR());
+
+           connection.response.subscribe((res: Response) => {
+             expect(res.statusText).toBe('');
```
Comment (jeffbcross) [inline modules/angular2/test/http/backends/xhr_backend_spec.ts]: Could we make the default status text (in BaseResponseOptions.ts) `OK`, consistent with the [fetch spec](https://fetch.spec.whatwg.org/#responses)?

**Comment by jeffbcross** [comment]
@lifus looks good to me, I just left a comment about setting the default value to "OK".

**Comment by lifus** [comment]
@jeffbcross sure, I'll fix that
BTW, [statusText defaults to empty string in angular.js](https://github.com/angular/angular.js/blob/d077966ff1ac18262f4615ff1a533db24d4432a7/src/ng/httpBackend.js#L85). [Here's a relevant test](https://github.com/angular/angular.js/blob/d077966ff1ac18262f4615ff1a533db24d4432a7/test/ng/httpBackendSpec.js#L84-L93).

**Comment by user414** [comment]
There was a few issues with this statusText, and statusCode being wrong. Posting on this since it seems to be the one being worked on and they could all be related. In addition, to this statusText, if you do a request and get a connection refused you still hand up getting a 200 response code.

**Comment by user414** [comment]
Referencing this issue here since they both changing the same section of the code and both waiting to be merged into master

https://github.com/angular/angular/issues/6390

**Pull request closed without merging**